physionet-django/training/views.py

```diff
@@ -0,0 +1,98 @@
+import json
+import operator
+from itertools import chain
+
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required, permission_required
+from django.db import transaction
+from django.db.models import Prefetch
+from django.http import JsonResponse
+from django.shortcuts import get_object_or_404, redirect, render
+from django.utils import timezone
+from django.utils.crypto import get_random_string
+
+from rest_framework.parsers import JSONParser
+
+from user.models import Training, TrainingType, TrainingQuestion, RequiredField
+from user.enums import TrainingStatus
+
+from training.models import Course, Quiz, QuizChoice, ContentBlock
+from training.models import CourseProgress, ModuleProgress, CompletedContent, CompletedQuiz
+from training.serializers import TrainingTypeSerializer
+
+
+@permission_required('training.change_course', raise_exception=True)
+def courses(request):
+    if request.POST:
+
+        if request.POST.get('training_id') != "-1":
+            training_type = get_object_or_404(TrainingType, pk=request.POST.get('training_id'))
+        else:
+            training_type = None
+
+        json_file = request.FILES.get("json_file", "")
+
+        if not json_file.name.endswith('.json'):
+            messages.warning(request, 'File is not of JSON type')
+            return redirect("courses")
+
+        # Checking if the content of the JSON file is properly formatted and according to the schema
+        try:
+            file_data = JSONParser().parse(json_file.file)
+        except json.decoder.JSONDecodeError:
+            messages.error(request, 'JSON file is not properly formatted.')
+            return redirect("courses")
+
+        # Checking if the Training type with the same version already exists
+        existing_course = Course.objects.filter(training_type=training_type)
+        if existing_course.exists():
+            # checking if the version number is valid
+            if not all(map(lambda x: x.isdigit() or x == '.', str(file_data['courses'][0]['version']))):
+                messages.error(request, 'Version number is not valid.')
+            # checking if the version number is greater than the existing version
+            elif float(file_data['courses'][0]['version']
+                       ) <= float(existing_course.order_by('-version').first().version):
+                messages.error(request, 'Version number should be greater than the existing version.')
```

Comment:
To ensure that the new version number is greater than the previous version, I am using a float based comparision, which does not work if the version has multiple decimal counters. Any suggestions on how we can compare that the new version is greater than the existing one?

Reply:
As mentioned above, there should be existing code that implements versioning for projects. I would suggest taking a look at that.

Reply:
See: https://github.com/MIT-LCP/physionet-build/pull/2068

Reply:
I've implemented the sematic package versioning used by python for version check.